docs/docs-new/pages/product/configuration/visualization-tools/langchain.mdx

```diff
@@ -0,0 +1,28 @@
+# Langchain
+
+[LangChain][langchain] is a framework designed to simplify the creation of
+applications using large language models.
+
+To get started with Langchain, follow the [instructions
+here][langchain-docs-getting-started].
+
+## Configuring the connection to Cube
+
+LangChain connects to Cube using the [REST API][ref-rest-api], and will need a
+[JWT][ref-jwt] to authenticate.
```

Comment:
I'm not sure this is exactly correct. LangChain doesn't connect to Cube. `CubeSemanticLoader`, a document loader we've contributed to LangChain, does.

I think this blog post should be linked from the docs page: https://cube.dev/blog/introducing-the-langchain-integration

Also, this language from the blog post seems relevant and can be copy-pasted (with light rewrites) to the docs:

> ...Cube's integration with LangChain. It comes as the [document loader](https://python.langchain.com/docs/integrations/document_loaders/cube_semantic) that is intended to be used to populate a vector database with embeddings derived from the data model of the semantic layer. Later, this vector database can be queried to find best-matching entities of the semantic layer. This is useful to match free-form input, e.g., queries in a natural language, with the views and their members in the data model.

![](https://ucarecdn.com/32e98c8b-a920-4620-a8d2-05d57618db8e/)

We're also providing an chat-based demo application (see [source code](https://github.com/cube-js/cube/tree/master/examples/langchain) on GitHub) with example [OpenAI prompts](https://github.com/cube-js/cube/blob/master/examples/langchain/utils.py#L48,L87) for constructing queries to Cube's [SQL API](https://cube.dev/docs/product/apis-integrations/sql-api). If you wish to create an [AI-powered conversational interface](https://cube.dev/blog/conversational-interface-for-semantic-layer) for the semantic layer, functioning similar to [Delphi](https://www.delphihq.com/), these prompts can be a good starting point.